websites/M/MarineTraffic/presence.ts

```diff
@@ -0,0 +1,330 @@
+		terms: { state: "Terms" },
+		"privacy-policy": { state: "Privacy Policy" },
+		"maritime-companies/search": { state: "Maritime companies search" },
+		"expand-coverage": { state: "Ship Spotting AIS Receiver" },
+		"apply-for-free-ais-receiver": { state: "Free AIS Receiver" },
+		"report-your-own-position": { state: "Report AIS Position" },
+		"online-services/plans": { state: "Online Services Plans" },
+		"photos/of/ships": { state: "Vessel photos" },
+		"ais/index/photographers": { state: "Photographers list" },
+		"users/my_account": { state: "User Account Overview" },
+		"api/account": { state: "API Services" },
+		"payment/billing": { state: "Billing Settings" },
+		"stations/index": { state: "Personal Stations" },
+		"fleets/index": { state: "User Fleets" },
+		"users/plans": { state: "Online Services & Plans" },
+		"maritime-companies/index": { state: "Business Profile" },
+		"corporate/index": { state: "Corporate Account" },
+		"users/settings": { state: "User Settings" },
+		"notifications/index": { state: "User Notifications" },
+	};
+
+	for (const [path, data] of Object.entries(pages)) {
+		if (document.location.href.includes(path))
+			presenceData = { ...presenceData, ...data };
+	}
+
+	if (document.location.hostname.startsWith("shop")) {
+		presenceData.details = "Browsing AIS e-shop:";
+		presenceData.state = document.title.split("-")[0];
+	} else if (document.location.hostname.startsWith("help")) {
+		presenceData.details = "Reading help:";
+		presenceData.state = document.title.split("–")[0];
+		presenceData.buttons = [
+			{ label: "Read this page", url: document.location.href },
+		];
+	} else if (document.location.pathname.startsWith("/blog")) {
+		presenceData.details = "Reading a blog post:";
+		presenceData.state = document.title.split("-")[0];
+		presenceData.buttons = [
+			{ label: "Read this post", url: document.location.href },
+		];
+	} else if (document.location.pathname.includes("ais/home/center")) {
+		presenceData.details = "Viewing live map";
+		setInterval(() => {
+			locationMap = document
+				.querySelector(
+					"#map_canvas > div.leaflet-control-container > div.leaflet-top.leaflet-right.leaflet-mouse-position-control > div.leaflet-control-mouseposition.leaflet-control"
+				)
+				.innerHTML.split("<br>")[0];
+		}, 5000);
+		presenceData.state = `Near ${locationMap}`;
+		presenceData.buttons = [
+			{ label: "View Page", url: document.location.href },
+		];
+	} else if (document.location.pathname.includes("ais/home/shipid")) {
+		elemName = capitalize(
+			document
+				.querySelector(
+					"#app > div > div.jss2 > div.d-flex.align-items-start.justify-content-start.flex-1-1-auto.home-layout > div.d-flex.flexcol.align-items-start.justify-content-start.flex-wrap.flex-1-1-auto > div > div:nth-child(4) > div > div > div > div > div > div.handle.cursor-move > div > div > div.d-flex.flexcol.align-items-start.justify-content-start.flex-wrap.flex-1-1-auto > div > a"
+				)
+				.innerHTML.toLowerCase()
```

Comment:
Try to get shorter selectors like this one:
![image](https://user-images.githubusercontent.com/42322979/187644710-4de11c0d-0e81-4294-87d9-48be3c77cbf4.png)
```suggestion
			document
				.querySelector(
					'[href*="/en/ais/details/ships/shipid:"]'
				).textContent.toLowerCase()
```